redis/asyncio/connection.py

```diff
@@ -817,7 +817,7 @@ def _error_message(self, exception: BaseException) -> str:
         else:
             return (
                 f"Error {exception.args[0]} connecting to {host_error}. "
-                f"{exception.args[0]}."
+                f"{exception}."
```

Comment:
Thanks for straightening this out.

This rendering of the error message is done in four places, sync and async, unix and network socket. All four slightly different from each other.

Would you care aligning all four places? Some ideas that come on the spot:
- pull out the handling at least into the two abstract connection classes, maybe even reuse a helper method between the two abstract classes;
- if there are no args, return a simple error message, don't even assume it's the "connection reset", there might be other cases too;
- if there is one arg, use that one;
- if there are more then one args, use the first one like it is today, and then put `exception.arg[1:]` at the end of the message, concatenated.

What do you think, does it make sense?

Reply:
I tried to push a commit with my suggestion, but I don't have permissions. I'll merge this as it is, it's already a good step forward. And open a separate PR as continuation.